src/compact/compact.js

```diff
@@ -0,0 +1,40 @@
+import { forEach } from "../for-each/for-each"
+import { is } from "../is/is"
+import { map } from "../map/map"
+import { filter } from "../filter/filter"
+import { curry } from "../curry/curry"
+
+const _compactObject = source => {
+  const result = { ...source }
+
+  forEach(key => {
+    if (!is(result[key])) {
+      delete result[key]
+    }
+  }, Object.keys(result))
+
+  return result
+}
+
+const _compactArray = source => filter(is, source)
+
+const _compact = source => {
+  const type = Array.isArray(source) ? "array" : typeof source
+
+  switch (type) {
+    case "array":
+      return _compactArray(source)
+
+    case "object":
+      return _compactObject(source)
+
+    default:
+      return source
+  }
+}
+
+const _compactMany = source => map(item => _compact(item), source)
+
+export const compact = curry(_compact)
```

Comment:
You don't need `curry` since `compact` and `compactMany` signatures have only one parameter.

Also, a comment block would help a lot, for both `compact` and `compactMany`, here's a starter:

```js
/**
 * Remove all "nothing" values from arrays or objects.
 *
 * @param {Array|Object} source
 *
 * @returns {Array|Object}
 *
 * @name compact
 * @tag Array,Object
 * @signature (source: Array|Object): Array|Object
 *
 * @see {@link compactMany}
 * @see {@link is}
 *
 * @example
 * compact([1, null, {}])
 * // => [1, {}]
 *
 * compact({
 *   a: "Lorem Ipsum",
 *   b: null,
 *   c: undefined,
 *   d: false,
 *   f: lambda,
 *  }),
 * // => {
 * //  a: "Lorem Ipsum",
 * //  d: false,
 * //  f: lamda
 * // }
 */
```